pkg/database/resync.go

```diff
@@ -77,115 +104,107 @@ func (dao *DAO) resetResources(ctx context.Context, resources []model.Resource,
 				continue
 			}
 
+			incomingResource, exists := incomingResMap[id]
+			if !exists {
+				// Resource needs to be deleted.
+				query, params, err := useGoqu(
+					"DELETE from search.resources WHERE uid IN ($1)",
```

Comment:
This sends a delete query per resource. Before all resources were deleted with a single query.
Need to measure the impact of this change. 
To test you can alternate sending 100k and 150k payloads.

Reply:
Also, I noticed that the existing query to delete all resources causes a significant increase in memory. I want to take a closer look at why.

Reply:
Yep this case is actually a regression. I'll revert this particular case of slice usage back to appending the ids to resourcesToDelete.

Reply:
Test data showing single query resourcesToDelete better than multi query: https://docs.google.com/document/d/1SxFi6ixU0SnvbHjJ829dzLtBZ0INGWXR3YoplVvjOJU/edit?tab=t.0#heading=h.ixyhqmahhf3r

Reply:
First though on large memory utilization of delete resources query would be leftover items in dao batch queue from previous updating of resources. Tried dropping batch size from 2500 -> 250, no discernible improvement.